### PR TITLE
Adjust to consul api 1.10.12

### PR DESCRIPTION
**Pull request by malfoj89**

### CODE REVIEW

File: src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcher.java [hunk 1]
```diff
@@ -48,13 +52,18 @@ public ServiceInstances decode(String s) {
...
-                    })
+                        try {
+                            return new ServiceInstance(
+                                    requiredValue(service, "ID", String.class),
+                                    requiredValue(service, "Tags", List.class),
+                                    requiredValue(service, "Address", String.class),
+                                    requiredValue(service, "Port", Integer.class)
+                            );
+                        } catch (Exception e) {
+                            logger.error(e.getMessage(), e.getCause());
```
Comment (matb4r) [inline src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcher.java:63]: Czy ten log bedzie nas informowal też o jaką konkretnie instancje/service chodzi? Mysle, ze dobrze by bylo to miec w logach

File: src/test/groovy/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcherIntTest.groovy [hunk 2]
```diff
@@ -34,13 +35,19 @@ class HealthServiceInstancesWatcherIntTest extends Specification {
     private EndpointWatcher<Services> healthServiceInstancesWatcher = recipes.healthServiceInstancesWatcher("my-service",
             recipes.consulWatcher(Executors.newFixedThreadPool(1))
                     .withBackoff(100, 1000)
-                    .build())
+                    .build()) as EndpointWatcher<Services>
+
+    def cleanup() {
+        consulCluster.findAllServices("dc1", "node1-dc1").getValue().forEach { key, val ->
```
Comment (matb4r) [inline src/test/groovy/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcherIntTest.groovy:41]: I would put this to separate method with clear name (like "deregisterServices"), and extract this strings to consts

File: src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcher.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (malfoj89) [reply src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcher.java:63]: valid issue, but we do not have this information granted there, and possibly service also could be null (or missing)

File: src/test/groovy/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcherIntTest.groovy (hunk 2 above)
Comment (malfoj89) [reply src/test/groovy/pl/allegro/tech/discovery/consul/recipes/watch/health/HealthServiceInstancesWatcherIntTest.groovy]: i though about it also, so if we both do I changed it as you pointed

**Comment by malfoj89** [comment]
this pr is invalid as it breaks solid

**Pull request closed without merging**